### ISSUE TITLE
Text as prompts

### ISSUE THREAD

**Issue by peiwang062** [facebookresearch/segment-anything#93]
Thanks for leasing this wonderful work!
I saw the demo shows examples of using point, box as input prompt. Does the demo support text as prompt?

**Comment by stefanjaspers** [comment]
Following! Text prompting has been mentioned in the research paper but hasn't been released yet. Really looking forward to this feature because I need it for a specific use case.

**Comment by darvilabtech** [comment]
Exactly, wait for it to be released

**Comment by HaoZhang990127** [comment]
Thank you for your exciting work!

I also want to use text as prompt to generate mask in my project. Now i am using ClipSeg to generate the mask, but it can not performance well in fine grained semantics.

When do you plan to open source the code of text as prompt? What is the approximate time line? Waiting for this amazing work.

**Comment by jy00161yang** [comment]
following

**Comment by eware-godaddy** [comment]
following

**Comment by jasonborn0** [comment]
The paper mentioned they used CLIP to handle text prompts: 
> We represent points and boxes by positional encodings [95] summed with learned embeddings for each prompt type and free-form text with an off-the-shelf text encoder from CLIP [82].

It appears the demo does not seem to allow textual inputs though.

**Comment by darvilabtech** [comment]
@peiwang062 @stefanjaspers @HaoZhang990127 @eware-godaddy @jy00161yang 
https://github.com/IDEA-Research/Grounded-Segment-Anything
Does what we are all looking after.

**Comment by peiwang062** [comment]
> @peiwang062 @stefanjaspers @HaoZhang990127 @eware-godaddy @jy00161yang https://github.com/IDEA-Research/Grounded-Segment-Anything does what we are all looking after.

yes, we could simply combine these two, but if SAM can do it better, why do we need two models. We don’t know if Grounding Dino is the bottleneck if we just use its output to SAM.

**Comment by alexw994** [comment]
> @peiwang062 @stefanjaspers @HaoZhang990127 @eware-godaddy @jy00161yang https://github.com/IDEA-Research/Grounded-Segment-Anything does what we are all looking after.

why not use output of SAM as bounding box?

**Comment by narbhar** [comment]
> > @peiwang062 @stefanjaspers @HaoZhang990127 @eware-godaddy @jy00161yang https://github.com/IDEA-Research/Grounded-Segment-Anything does what we are all looking after.
> 
> why not use output of SAM as bounding box?

The current version of SAM without of the CLIP text encoder only gives instances from image points or bounding boxes as prompts. Thus SAM's output is only instances without semantic information about the segmentation. With a text encoder you can correlate SAM's output with text such as an object of interest in an image. If you have a promptable text based object detector you can bridge this link for the time being thus SAM's output not generic instance segmentation anymore which is what Grounded Segment Anything is helping to do.

**Comment by luca-medeiros** [comment]
Put together a demo of grounded-segment-anything with radio for better testing.
I tested using clip, open-clip, and groundingdino. Groudingdino performs much better with a great performance. Less than 1 sec on a A100 for DINO+SAM. Maybe ill add the clip versions as well.
https://github.com/luca-medeiros/lang-segment-anything

**Comment by alexw994** [comment]
> > > @peiwang062 @stefanjaspers @HaoZhang990127 @eware-godaddy @jy00161yang https://github.com/IDEA-Research/Grounded-Segment-Anything does what we are all looking after.
> > 
> > 
> > why not use output of SAM as bounding box?
> 
> The current version of SAM without of the CLIP text encoder only gives instances from image points or bounding boxes as prompts. Thus SAM's output is only instances without semantic information about the segmentation. With a text encoder you can correlate SAM's output with text such as an object of interest in an image. If you have a promptable text based object detector you can bridge this link for the time being thus SAM's output not generic instance segmentation anymore which is what Grounded Segment Anything is helping to do.


If SAM just segments in the bounding box, I think many methods can be used for this also. Like BoxInstSeg https://github.com/LiWentomng/BoxInstSeg

**Comment by peiwang062** [comment]
> > > > @peiwang062 @stefanjaspers @HaoZhang990127 @eware-godaddy @jy00161yang https://github.com/IDEA-Research/Grounded-Segment-Anything does what we are all looking after.
> > > 
> > > 
> > > why not use output of SAM as bounding box?
> > 
> > 
> > The current version of SAM without of the CLIP text encoder only gives instances from image points or bounding boxes as prompts. Thus SAM's output is only instances without semantic information about the segmentation. With a text encoder you can correlate SAM's output with text such as an object of interest in an image. If you have a promptable text based object detector you can bridge this link for the time being thus SAM's output not generic instance segmentation anymore which is what Grounded Segment Anything is helping to do.
> 
> If SAM just segment in the bounding box, I think many methods can be used for this also. Like BoxInstSeg https://github.com/LiWentomng/BoxInstSeg

It should be able to support boxes, points, masks and text as prompts as the paper mentions, no?

**Comment by nikolausWest** [comment]
following

**Comment by yash0307** [comment]
following

**Comment by 9p15p** [comment]
Following

**Comment by Zhangwenyao1** [comment]
following

**Comment by Eli-YiLi** [comment]
Our work can achieve text to mask with SAM:
https://github.com/xmed-lab/CLIP_Surgery

This is our work about CLIP's explainability. It's able to guide SAM to achieve text to mask without manual points. 

Besides, it's very simple without any fine-tuning, using the CLIP model itself only. 

Furthermore, it enhances many open-vocabulary tasks, like segmentation, multi-label classification, multimodal visualization. 

This is the jupyter demo:
https://github.com/xmed-lab/CLIP_Surgery/blob/master/demo.ipynb

![fig4](https://user-images.githubusercontent.com/49093246/231662220-2b91306c-4dc8-42d1-9fdd-3fbbafadeb26.jpg)

**Comment by zaojiahua** [comment]
following

**Comment by FrancisDacian** [comment]
can try the result using this explorer extension: 
[https://chrome.google.com/webstore/detail/text-prompts-for-segment/jndfmkiclniflknfifngodjnmlibhjdo/related

**Comment by FrancisDacian** [comment]
can try the result using this explorer extension:
[](https://chrome.google.com/webstore/detail/text-prompts-for-segment/jndfmkiclniflknfifngodjnmlibhjdo/related)

**Comment by FrancisDacian** [comment]
can try the result using this explorer extension
[https://chrome.google.com/webstore/detail/text-prompts-for-segment/jndfmkiclniflknfifngodjnmlibhjdo/related](url)

**Comment by ignoHH** [comment]
following

**Comment by bjccdsrlcr** [comment]
following

**Comment by mydcxiao** [comment]
following

**Comment by xuxiaoxxxx** [comment]
+1

**Comment by daminnock** [comment]
following

**Comment by Alice1820** [comment]
following

**Comment by freshman97** [comment]
following

**Comment by zhangjingxian1998** [comment]
waiting for it

**Comment by N-one** [comment]
following

**Comment by moktsuiqin** [comment]
following
